java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -284,9 +284,11 @@ private AccessToken refreshAccessToken(int attempt) throws IOException {
                     tokenResponse = exchangeRefreshTokenForAccessToken();
                     token = asAccessToken(tokenResponse);
 
-                    storeSharedAccessTokenIfSupported(token);
                     if (isAccessTokenCacheable()) {
                         this.cachedToken = token;
+                    } else {
+                        //q: assuming that if the token is cacheable, is not present as a SECRET to be stored
+                        storeSharedAccessTokenIfSupported(token);
```

Comment:
That will fail if the secret token is not defined in secret manager / parameter store. I think your idea is to get rid off that if token is cacheable? 

This is the main change done apart from fixes.

Reply:
sry, yeah - if we have `USE_SHARED_TOKEN`, then we need to define `ACCESS_TOKEN` secret - yes, I agree.